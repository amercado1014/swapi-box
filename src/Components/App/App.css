.title-img {
  display: block;
  margin: 25px auto;
  width: 50%;
}

