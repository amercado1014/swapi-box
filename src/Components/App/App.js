import React, { Component } from 'react';
import Nav from '../Nav/Nav';
import CardContainer from '../CardContainer/CardContainer';
import Summary from '../Summary/Summary';
import Swapi from '../../api';
import './App.css';

class App extends Component {
  constructor() {
    super();

    this.state = {
      film: {},
      cards: [],
      people: [],
      planets: [],
      vehicles: [],
      favorites: [],
      errorStatus: false
    };
  }

  async componentDidMount() {
    await this.fetchApiData('film', 'filmData');
  }
  
  fetchApiData = async (category, dataCategory) => {
    if (!this.state[category].length) {
      try {
        const fetchedData = await Swapi[dataCategory]();
        this.setState({[category]: fetchedData});
      } catch (error) {
        this.setState({ errorStatus: true });
      }
    } 
    this.setCardsArray(category);  
  }

  setCardsArray = (category) => {
    if (category !== 'film') {
      this.setState({ cards: this.state[category] });
    }
  }

  setFavoritesArray = (card) => {
    const found =
      this.state.favorites.find(favorite => favorite.name === card.name);
    if (!found) {
      this.setState({
        favorites: [...this.state.favorites, card]
      });
    } else {
      const filtered =
        this.state.favorites.filter(favorite => favorite.name !== card.name);
      this.setState({
        favorites: filtered
      });
    }
  }

  render() {
    const { film, cards, favorites } = this.state;
    return (
      <div className="App"> 
        <img className='title-img' 
          src={require('../../swapi.png')} 
          alt="swapi logo"/>
        <Nav 
          fetchCards={this.fetchApiData} 
          favorites={favorites} 
          displayFavorites={this.setCardsArray} />
        <Summary film={film}/>
        <CardContainer 
          cards={cards} 
          setFavorites={this.setFavoritesArray}
          favorites={favorites} /> 
      </div>
    );
  }
}

export default App;
