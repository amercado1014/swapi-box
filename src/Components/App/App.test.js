import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
