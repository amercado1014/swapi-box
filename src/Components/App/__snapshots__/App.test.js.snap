// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
>
          SWAPI-Box
</h1>,
        <div
          className="flex"
>
          <Summary
                    film={Object {}}
          />
          <div
                    className="right-side"
          >
                    <Nav
                              displayFavorites={[Function]}
                              favorites={Array []}
                              fetchCards={[Function]}
                    />
                    <CardContainer
                              cards={Array []}
                              setFavorites={[Function]}
                    />
          </div>
</div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SWAPI-Box",
          "className": "title",
        },
        "ref": null,
        "rendered": "SWAPI-Box",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Summary
              film={Object {}}
/>,
            <div
              className="right-side"
>
              <Nav
                            displayFavorites={[Function]}
                            favorites={Array []}
                            fetchCards={[Function]}
              />
              <CardContainer
                            cards={Array []}
                            setFavorites={[Function]}
              />
</div>,
          ],
          "className": "flex",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "film": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Nav
                  displayFavorites={[Function]}
                  favorites={Array []}
                  fetchCards={[Function]}
/>,
                <CardContainer
                  cards={Array []}
                  setFavorites={[Function]}
/>,
              ],
              "className": "right-side",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "displayFavorites": [Function],
                  "favorites": Array [],
                  "fetchCards": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "cards": Array [],
                  "setFavorites": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
>
            SWAPI-Box
</h1>,
          <div
            className="flex"
>
            <Summary
                        film={Object {}}
            />
            <div
                        className="right-side"
            >
                        <Nav
                                    displayFavorites={[Function]}
                                    favorites={Array []}
                                    fetchCards={[Function]}
                        />
                        <CardContainer
                                    cards={Array []}
                                    setFavorites={[Function]}
                        />
            </div>
</div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SWAPI-Box",
            "className": "title",
          },
          "ref": null,
          "rendered": "SWAPI-Box",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Summary
                film={Object {}}
/>,
              <div
                className="right-side"
>
                <Nav
                                displayFavorites={[Function]}
                                favorites={Array []}
                                fetchCards={[Function]}
                />
                <CardContainer
                                cards={Array []}
                                setFavorites={[Function]}
                />
</div>,
            ],
            "className": "flex",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "film": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Nav
                    displayFavorites={[Function]}
                    favorites={Array []}
                    fetchCards={[Function]}
/>,
                  <CardContainer
                    cards={Array []}
                    setFavorites={[Function]}
/>,
                ],
                "className": "right-side",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "displayFavorites": [Function],
                    "favorites": Array [],
                    "fetchCards": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "cards": Array [],
                    "setFavorites": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
